tests/flow/test_function_calls.py

```diff
@@ -335,3 +336,31 @@ def test16_edge_endpoints(self):
             self.env.assertEquals(row[0], row[1])
             self.env.assertEquals(row[2], row[3])
 
+    def test17_to_json(self):
+        # Test JSON literal values in an array.
+        query = """RETURN toJSON([1, 'str', true, NULL])"""
+        actual_result = graph.query(query)
+        parsed = json.loads(actual_result.result_set[0][0])
+        self.env.assertEquals(parsed, [1, "str", True, None])
+
+        # Test converting a map projection.
+        query = """MATCH (n {val: 1}) RETURN toJSON(n {.val, .name})"""
+        actual_result = graph.query(query)
+        parsed = json.loads(actual_result.result_set[0][0])
+        self.env.assertEquals(parsed, {"name": "Alon", "val": 1})
+
+        # Test converting a full node.
+        query = """RETURN toJSON([1, 'str', true, NULL])"""
+        query = """MATCH (n {val: 1}) RETURN toJSON(n)"""
+        actual_result = graph.query(query)
+        parsed = json.loads(actual_result.result_set[0][0])
+        self.env.assertEquals(parsed, {"type": "node", "id": 1, "labels": ["person"], "properties": {"name": "Alon", "val": 1}})
+
+        # Test converting a full edge.
+        query = """MATCH ({val: 0})-[e:works_with]->({val: 1}) RETURN toJSON(e)"""
+        actual_result = graph.query(query)
+        start = {"id": 0, "labels": ["person"], "properties": {"name": "Roi", "val": 0}}
+        end = {"id": 1, "labels": ["person"], "properties": {"name": "Alon", "val": 1}}
+        parsed = json.loads(actual_result.result_set[0][0])
+        self.env.assertEquals(parsed, {"type": "relationship", "id": 12, "relationship": "works_with", "properties": {}, "start": start, "end": end})
+
```

Comment:
Test the `toJSON` on empty composite types (array, path and map)
in addition test this function on a path with just a single node.